File: core/src/org/rexellentgames/dungeon/entity/item/accessory/equipable/VampireRing.java
package org.rexellentgames.dungeon.entity.item.accessory.equipable;

import org.rexellentgames.dungeon.assets.Locale;
import org.rexellentgames.dungeon.entity.creature.player.Player;

public class VampireRing extends Equipable {
	{
		description = Locale.get("vampire_ring_desc");
		name = Locale.get("vampire_ring");
	}

	@Override
	public void onEquip() {
		super.onEquip();

		if (this.owner instanceof Player) {
			((Player) this.owner).vampire += 5f;
		}
	}

	@Override
	public void onUnequip() {
		super.onUnequip();

		if (this.owner instanceof Player) {
			((Player) this.owner).vampire -= 5f;
		}
	}
}

File: core/src/org/rexcellentgames/burningknight/entity/item/accessory/equippable/VampireRing.kt
package org.rexcellentgames.burningknight.entity.item.accessory.equippable

import org.rexcellentgames.burningknight.assets.Locale

class VampireRing : Equippable() {
	init {
		description = Locale.get("vampire_ring_desc")
		name = Locale.get("vampire_ring")
		sprite = "item-ring_j"
	}

	override fun onEquip(load: Boolean) {
		super.onEquip(load)
		this.owner.vampire += getChance()
	}

	override fun getMaxLevel(): Int {
		return 10
	}

	override fun canBeDegraded(): Boolean {
		return false
	}

	override fun onUnequip(load: Boolean) {
		super.onUnequip(load)
		this.owner.vampire -= getChance()
	}

	private fun getChance(): Float {
		return this.level * 10f
	}

	override fun getDescription(): String {
		return super.getDescription().replace("{CHANCE}", getChance().toInt().toString())
	}
}